### ISSUE TITLE
salt-cloud: Generating traceback when exception is thrown by provider

### ISSUE THREAD

**Issue by syphernl** [saltstack/salt#19156]
It appears that salt-cloud doesn't like it when a cloud provider raises an exception:
```
[ERROR   ] Failed to deploy 'vmtest2'. Error:
[ERROR   ] Caught Exception, terminating workers
TRACE: 'bool' object has no attribute 'pop'
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 58, in _call
    ret = func(*args, **kw)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 2129, in create_multiprocessing
    output.pop('deploy_kwargs', None)
AttributeError: 'bool' object has no attribute 'pop'



Error: There was a query error: Exception caught
Caught Exception, terminating workers
TRACE: 'bool' object has no attribute 'pop'
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 58, in _call
    ret = func(*args, **kw)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 2129, in create_multiprocessing
    output.pop('deploy_kwargs', None)
AttributeError: 'bool' object has no attribute 'pop'
```

Version details:
```
            Salt: 2014.7.0
          Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
          Jinja2: 2.7.2
        M2Crypto: 0.21.1
  msgpack-python: 0.3.0
    msgpack-pure: Not Installed
        pycrypto: 2.6.1
         libnacl: Not Installed
          PyYAML: 3.10
           ioflo: Not Installed
           PyZMQ: 14.0.1
            RAET: Not Installed
             ZMQ: 4.0.4
            Mako: 0.9.1
 Apache Libcloud: 0.14.1
```

**Comment by ssgward** [comment]
Is this still an issue on the latest version of salt? If so, can updated info be added to the bug to be current? Otherwise, close the bug.

**Comment by rallytime** [comment]
Since we haven't heard back here, I am going to close this bug as there have been numerous fixes in this area since 2014.7.0. If this remains an issue on a newer version of salt (2015.5.1 is current stable) please leave a comment and we will gladly re-open this bug and take another look. Thanks!
